### ISSUE TITLE
no-underscore-dangle to check function params

### ISSUE THREAD

**Issue by hnipps** [eslint/eslint#12579]
**What rule do you want to change?**
no-underscore-dangle

**Does this change cause the rule to produce more or fewer warnings?**
More warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
The updated rule will check if param Identifiers of FunctionDeclarations and ArrowFunctionExpressions have dangling underscores. 

I suggest that underscore dangle in these situations is disallowed by default and can be configured using the same pattern as `"allowAfterThis"` and `"allowAfterSuper"`.

**Please provide some example code that this change will affect:**

```js
const test = (_arg) => { };
function myfunc(_arg) { };
```

**What does the rule currently do for this code?**
No error thrown.

**What will the rule do after it's changed?**
Throw an error because the function params have dangling underscores.

**Are you willing to submit a pull request to implement this change?**
Yes

**Comment by hnipps** [comment]
This update will help users migrating from TSLint to ESLint to reach feature parity because the TSLint [`variable-name`](https://palantir.github.io/tslint/rules/variable-name/) rule already has the proposed feature.

**Comment by webOS101** [comment]
I would suggest this be configurable, as well, or, perhaps allow a single underscore in parameter names by default, given the common pattern of using `_` for unused parameter.

**Comment by kaicataldo** [comment]
I agree that this seems like an oversight of the rule and support this change. I do think it needs to be put behind an option to avoid a breaking change (maybe `allowFunctionParams: true` by default) and that in addition to `FunctionDeclaration`s and `ArrowFunctionExpression`s, I think it should also check `FunctionExpression`s.

**Comment by hnipps** [comment]
Is this still being evaluated? Or do we have a decision?

**Comment by kaicataldo** [comment]
I'll champion this. We still need support from two other team members before we can mark this as accepted.

**Comment by ron23** [comment]
I would love to see that as well. We're migrating from TSLint and would really like to keep the consistency. Thanks!

**Comment by anikethsaha** [comment]
Marking this as accepted as there are 3 +1 from team members. 

I think `allowFunctionParams` might be a good approach and name which will be `true` by default as mentioned above
